warehouse/query-core/src/main/java/datawave/query/jexl/functions/TermFrequencyAggregator.java

```diff
@@ -60,4 +69,30 @@ protected Key getSeekStartKey(Key current, ByteSequence pointer) {
         // seek to the next documents TF
         return new Key(current.getRow(), current.getColumnFamily(), new Text(pointer + Constants.NULL_BYTE_STRING + Constants.MAX_UNICODE_STRING));
     }
+    
+    @Override
+    public Key apply(SortedKeyValueIterator<Key,Value> itr, Document doc, AttributeFactory attrs) throws IOException {
+        Key key = super.apply(itr, doc, attrs);
+        
+        boolean necessary = false;
+        // test that the returned document has values required for evaluation
+        for (Attribute attr : doc.getDictionary().values()) {
+            if (attr.isToKeep()) {
+                necessary = true;
+                break;
+            }
+        }
+        
+        // only return a key if something was added to the document necessary for evaluating the query
+        if (!necessary) {
```

Comment:
why is attr.isToKeep() the determination that it is necessary for evaluation.  The fact that it is in the document means it is necessary in the first place if the underlying filters did the correct thing.

Reply:
yep, this slipped through. I will remove it